scripts/migrate_duplicate_external_accounts.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import absolute_import
+
+import logging
+import sys
+
+from dropbox.rest import ErrorResponse
+from dropbox.client import DropboxClient
+
+from framework.mongo import database as db
+from framework.transactions.context import TokuTransaction
+from website.app import init_app
+from website.addons.github.api import GitHubClient
+from website.addons.github.exceptions import GitHubError
+from website.oauth.models import ExternalAccount
+from scripts import utils as script_utils
+
+
+logger = logging.getLogger(__name__)
+
+def creds_are_valid(ea_id):
+    ea = ExternalAccount.load(ea_id)
+    if ea.provider == 'github':
+        try:
+            GitHubClient(external_account=ea).user()
+        except (GitHubError, IndexError):
+            logger.info('Invalid creds: {}'.format(ea_id))
+            return False
+    elif ea.provider == 'dropbox':
+        try:
+            DropboxClient(ea.oauth_key).account_info()
+        except (ValueError, IndexError, ErrorResponse):
+            logger.info('Invalid creds: {}'.format(ea_id))
+            return False
+    else:
+        raise Exception('Unexpected provider: {}'.format(ea.provider))
+    logger.info('Valid creds: {}'.format(ea_id))
+    return True
+
+def swap_references_and_rm(to_keep, to_swap, provider):
+    logger.info('Swapping {} references to {} with {}'.format(provider, to_swap, to_keep))
+    db['{}nodesettings'.format(provider)].find_and_modify(
+        {'external_account': to_swap},
+        {'$set': {
+            'external_account': to_keep
+        }}
+    )
+    us_map = {us['_id']: us['external_accounts'] for us in db['{}usersettings'.format(provider)].find({'external_accounts': to_swap})}
```

Comment:
I don't think it's necessary to check usersettings:

```

In [11]: db.dropboxusersettings.find({'external_accounts': {'$exists': True}}).count()
Out[11]: 0

In [12]: db.githubusersettings.find({'external_accounts': {'$exists': True}}).count()
Out[12]: 0
```

Not a huge deal, though.